### PR TITLE
feature/whitelist

### PR DESCRIPTION
**Pull request by dovgopoly**

### CODE REVIEW

File: contracts/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@dlsl/dev-modules",
-  "version": "2.0.3",
+  "version": "2.1.3",
```
Comment (Arvolear) [inline contracts/package.json:3]: Why 2.1.3? Should be 2.1.0

File: contracts/access-control/MerkleWhitelisted.sol [hunk 2]
```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+
+/**
+ *  @notice The Whitelist Access Control module
+ *
+ *  This is the simple abstract contract that implements whitelisting logic.
```
Comment (Arvolear) [inline contracts/access-control/MerkleWhitelisted.sol:9]:
```suggestion
 *  This is a simple abstract contract that implements whitelisting logic.
```

File: contracts/access-control/MerkleWhitelisted.sol [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ *  It is based on the Merkle tree. This means that you first need to compute the Merkle tree off-chain
+ *  and set its root using `_setMerkleRoot` function from your child contract. You also need to perform
+ *  this action every time the state of the whitelist changes.
```
Comment (Arvolear) [inline contracts/access-control/MerkleWhitelisted.sol:13]:
```suggestion
 *  The contract is based on a Merkle tree, which allows for the huge whitelists to be cheaply validated courtesy of
 *  O(log(n)) tree complexity. The whitelist itself is stored in the tree leaves and only the root of the tree is saved on-chain. 
 *
 *  To validate the whitelist belonging, the tree leaf (the whitelist element) has to be computed and passed to the 
 *  "root-construction" function together with the corresponding tree branches. The function will then check the 
 *  roots equality and if they match, the element belongs to the list.
 * 
 *  Note: the branch nodes are sorted numerically.
```

File: contracts/package.json (hunk 1 above)
Comment (dovgopoly) [reply contracts/package.json]: done

File: contracts/access-control/MerkleWhitelisted.sol (hunk 2 above)
Comment (dovgopoly) [reply contracts/access-control/MerkleWhitelisted.sol]: done

File: contracts/access-control/MerkleWhitelisted.sol (hunk 3 above)
Comment (dovgopoly) [reply contracts/access-control/MerkleWhitelisted.sol]: done

**Pull request merged**